midend/removeReturns.cpp

```diff
@@ -132,7 +132,8 @@ const IR::Node* DoRemoveReturns::preorder(IR::BlockStatement* statement) {
             ret = r;
         }
     }
-    set(ret);
+    if (!stack.empty())
```

Comment:
Its not entirely clear that it is OK -- perhaps it should be pruning the preorder function for Delcaration_Instance instead.